pkg/logql/ast.go

```diff
@@ -364,6 +365,58 @@ func (e *LineFilterExpr) Filter() (log.Filterer, error) {
 
 	return f, nil
 }
+*/
+func (e *LineFilterExpr) Filter() (log.Filterer, error) {
+
+	acc := make([]log.Filterer, 0)
+	for curr := e; curr != nil; curr = curr.Left {
+
+		switch curr.Op {
+		case OpFilterIP:
+			var err error
+			next, err := log.NewIPLineFilter(curr.Match, curr.Ty)
+			if err != nil {
+				return nil, err
+			}
+			acc = append(acc, next)
+		default:
+			var err error // to avoid `f` being shadowed.
+			next, err := log.NewFilter(curr.Match, curr.Ty)
+			if err != nil {
+				return nil, err
+			}
+			acc = append(acc, next)
+		}
+	}
+
+	// Join all contain filters if there are any.
+	containsAllFilter := log.ContainsAllFilter{}
+	regexpFilters := make()
+	n := 0
+	for _, filter := range acc {
+		switch c := filter.(type) {
+		case log.ContainsFilter:
+			{
```

Comment:
I don't think we need an explicit block here. Is there something I'm missing?

Reply:
My Golang skills :laughing: